src/student.py

```diff
@@ -7,22 +7,65 @@ def __init__(self, student_id, subjects_and_classes, subject_targets, subject_gi
         self.buddies = buddies   #Dictionary where for each subject (key) there is a list of ups ordered by their priority
     
     def add_subject_and_class(self, subject_name, class_number):
-        self.subjects_and_classes[subject_name] = class_number
+        if len(self.subjects_and_classes) < 7:
```

Comment:
@tiago-falves where does this 7 come from?

Reply:
It is the maximum amout of classes a student can do in a semester!